### PR TITLE
docs: restructure Getting Started and Advanced Installation sections

### PR DESCRIPTION
**Pull request by yoyo-go**
According to the [restructure plan](https://docs.google.com/document/d/1gIc6odw6n3bPmO-3LCUiaJ8BmlSgVPug/edit), the Getting Started and Advanced Installation sections are updated, when testing the docs, below warnings are shown:

1. `advancedinstallation/rancher-desktop-configure.rst:5: WARNING: Include file '/src/Documentation/gettingstarted/rancher-desktop-override.yaml' not found or reading it failed`

2. `advancedinstallation/rancher-desktop-configure.rst:3: WARNING: download file not readable: /src/Documentation/gettingstarted/rancher-desktop-override.yaml`

3. `/src/Documentation/advancedinstallation/microk8s.rst: WARNING: document isn't included in any toctree`

4. `/src/Documentation/concepts/index.rst: WARNING: document isn't included in any toctree`

For the 1st and 2nd warnings, the `rancher-desktop-override.yaml` is already under the `advancedinstallation` directory, I don't know why it throws this warning.

For the 3rd and 4th warnings, the `microk8s` is retired and all the topics under concepts section will move to other directories. So can I leave this warnings here?

### CODE REVIEW

**Comment by yoyo-go** [comment]
Another thing is I have added the Signed-off lable in my commit, why the Bot still said I need a signed-off?
```
commit b1a5bc6a8d216a3564da9e3f48d3e05516f8eb88 (HEAD -> docs-restructure-gettingstarted, origin/docs-restructure-gettingstarted)
Author: yoyo-go <yoyo19980720@126.com>
Date:   Sat Jun 18 17:03:18 2022 +0800

    Update Getting Started and Advanced Installation sections

    Signed-off-by: Yoyo Wu <yoyo19980720@126.com>
```

**Comment by joestringer** [comment]
> Another thing is I have added the Signed-off lable in my commit, why the Bot still said I need a signed-off?

Every commit needs this tag. Currently the middle commit does a "merge". You should be able to `git rebase --interactive` against the latest master tree and remove this commit, then only the commits with your intended changes + signoff will remain.

**Comment by yoyo-go** [comment]
> Every commit needs this tag. Currently the middle commit does a "merge". You should be able to `git rebase --interactive` against the latest ~master~ `feature/gsod-docs-reorg` branch and remove this commit, then only the commits with your intended changes + signoff will remain.

When `git rebase -i`, the Merge commit didn't show up in the `pick xxx` list, only the first and the third commit can be actioned...

**Comment by sayboras** [comment]
> When git rebase -i, the Merge commit didn't show up in the pick xxx list

This is expected if you are `git rebase -i feature/gsod-docs-reorg` as the second commit is already available in feature/gsod-docs-reorg branch i.e. only your additional commits are shown in rebase list.

**Comment by qmonnet** [comment]
image

The above is because we still include `rancher-desktop-configure.rst` from `gettingstarted/k8s-install-default.rst`, so when we look for the YAML file, this is done relatively to the path of `k8s-install-default.rst` (i.e. `gettingstarted`):

```
$ git grep 'rancher-desktop-configure'
advancedinstallation/k8s-install-helm.rst:       .. include:: rancher-desktop-configure.rst
advancedinstallation/rancher-desktop.rst:.. include:: rancher-desktop-configure.rst
gettingstarted/k8s-install-default.rst:       .. include:: ../advancedinstallation/rancher-desktop-configure.rst
```

This should be fixable by using absolute paths:

``` diff
diff --git a/Documentation/advancedinstallation/rancher-desktop-configure.rst b/Documentation/advancedinstallation/rancher-desktop-configure.rst
index 49c1a33b97e9..91a565ef101a 100644
--- a/Documentation/advancedinstallation/rancher-desktop-configure.rst
+++ b/Documentation/advancedinstallation/rancher-desktop-configure.rst
@@ -2,9 +2,9 @@ Configuring Rancher Desktop is done using a YAML configuration file.
 This step is necessary in order to disable the default CNI and replace it with
 Cilium.
 
-Next you need to start Rancher Desktop with ``containerd`` and create a :download:`override.yaml <rancher-desktop-override.yaml>`:
+Next you need to start Rancher Desktop with ``containerd`` and create a :download:`override.yaml </advancedinstallation/rancher-desktop-override.yaml>`:
 
-.. literalinclude:: rancher-desktop-override.yaml
+.. literalinclude:: /advancedinstallation/rancher-desktop-override.yaml
    :language: yaml
 
 After the file is created move it into your Rancher Desktop's ``lima/_config`` directory:
```

File: Documentation/index.rst [hunk 1]
```diff
@@ -53,9 +53,22 @@ get started and experiment with Cilium.
...
+   gettingstarted/hubble_cli.rst
+   gettingstarted/terminology
+   gettingstarted/gettinghelp
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Advanced Installation
+
+   advancedinstallation/taints
+   advancedinstallation/index
```
Comment (qmonnet) [inline Documentation/index.rst]: Should `taints` and `index` be swapped here? Or should even `taints` go into a dedicated section in `advancedinstallation/index`?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Please address the remaining CI issue

image

@lizrice @xmulligan Are you OK with the current change? Is `advancedinstallation` OK for the directory name, or do we want something shorter (`advanced` is maybe not explicit enough, maybe `advancedinstall`)?

**Comment by yoyo-go** [comment]
> Please address the remaining CI issue
> 
> image
> 
> @lizrice @xmulligan Are you OK with the current change? Is `advancedinstallation` OK for the directory name, or do we want something shorter (`advanced` is maybe not explicit enough, maybe `advancedinstall`)?

The issues have been solved with the latest commit. Please have a look, thanks!

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good to me!
Can you please squash your three commits together?

**Comment by yoyo-go** [comment]
> Looks good to me! Can you please squash your three commits together?

I have squashed the 3 commits into 1 commit now.

**Review by qmonnet — APPROVED** [review]

**Pull request merged**